### PR TITLE
Fetch data V2

### PR DESCRIPTION
**Pull request by onadrog**
Tweaks again.
Since some games had no description due to regex sometimes return not a valid slug. 
I used GraphQl  to get the productSlug andd return it into the API endpoint.

cons: 2 call per gamePage instead of 1
Pros: now ~97% of games has description or shortDescription

I used a filter to get the right page.data object. Before it sometimes gave DLC description instead of real game.

### CODE REVIEW

**Review by flavioislima — COMMENTED** [review]
Looks good to me! 👍🏽 
Just be sure to fix the conflicts and to check if no translation was broken.
Would be good to run yarn i18n a check if nothing broke ⚔️

**Comment by onadrog** [comment]
Yes i'll rebase next-week and see for confilcts

File: electron/main.ts [hunk 1]
```diff
@@ -262,20 +262,53 @@ ipcMain.on('quit', async () => handleExit())
 /* const storage: Storage = mainWindow.localStorage
 const lang = storage.getItem('language') */
 
-ipcMain.handle('getGameInfo', async (event, game) => {
+const fetchGraphQl = async (namespace: string, game:string) => {
```
Comment (ianwijma) [inline electron/main.ts:265]: Function name is too Ambiguous. Having a more descriptive function name would make it easier to understand what the function does. Like getGameCatelogSlug or something like that.

File: electron/main.ts [hunk 2]
```diff
@@ -262,20 +262,53 @@ ipcMain.on('quit', async () => handleExit())
...
   let lang = JSON.parse(readFileSync(heroicConfigPath, 'utf-8')).defaultSettings
     .language
   if (lang === 'pt') {
     lang = 'pt-BR'
   }
-  const epicUrl = `https://store-content.ak.epicgames.com/api/${lang}/content/products/${game}`
+
+  let epicUrl:string;
+  if (namespace) {
+    const fetch:string = await fetchGraphQl(namespace, game)
```
Comment (ianwijma) [inline electron/main.ts:298]: Fetch is super ambiguous and could cause confusion upon reading.

File: electron/main.ts [hunk 3]
```diff
@@ -399,9 +432,8 @@ ipcMain.on('callTool', async (event, { tool, wine, prefix, exe }: Tools) => {
     winePrefix = `'${protonPrefix}/pfx'`
   }
 
-  let command = `WINE=${wineBin} WINEPREFIX=${winePrefix} ${
-    tool === 'winecfg' ? `${wineBin} ${tool}` : tool
-  }`
+  let command = `WINE=${wineBin} WINEPREFIX=${winePrefix} ${tool === 'winecfg' ? `${wineBin} ${tool}` : tool
+    }`
```
Comment (ianwijma) [inline electron/main.ts:436]: Weird code formatting on this like.

File: electron/legendary_utils/library.ts [hunk 4]
```diff
@@ -62,6 +62,8 @@ export async function getLegendaryConfig(file: string) {
             dlcItemList,
             customAttributes: { CloudSaveFolder, FolderName },
           } = metadata
+          
+          const {catalog_item_id, namespace} = asset_info
```
Comment (ianwijma) [inline electron/legendary_utils/library.ts:66]: I do not see `catalog_item_id` being used anywhere, what is the use / reason of it being added?

File: public/locales/pl/gamepage.json [hunk 5]
```diff
@@ -121,36 +121,37 @@
       "stop": "Gra włączona (Zatrzymaj)"
     },
     "saves": {
-      "syncing": "Synchronizacja zapisów gry"
-    "submenu": {
+      "syncing": "Synchronizacja zapisów gry",
+      "submenu": {
```
Comment (ianwijma) [inline public/locales/pl/gamepage.json:125]: pretty sure @flavioislima has fixed issues with the Polish translations, maybe a merge from the main branch will fix issues with these translations.

**Review by ianwijma — CHANGES_REQUESTED** [review]
Only found some minor issues. Note my comment on the polish file.

File: electron/main.ts (hunk 2 above)
Comment (flavioislima) [reply electron/main.ts:298]: Good point. Also fetch is a javascript function. Try something like getSomething... since this is an action.

**Comment by onadrog** [comment]
Fixes done.

**Review by flavioislima — APPROVED** [review]
Nice 👍

**Review by ianwijma — APPROVED** [review]
It looks good :+1:

**Pull request merged**